docker_compose/manage_secrets.py

```diff
@@ -0,0 +1,325 @@
+        worker_poller_file["services"]["worker-poller"]["secrets"] = list(
+            filter(
+                lambda el: el["source"] not in secrets,
+                worker_poller_file["services"]["worker-poller"]["secrets"],
+            )
+        )
+
+        # Save updated docker-compose-worker-poller.yaml configuration
+        with open(
+            os.path.join(path_to_compose_files, "docker-compose-worker-poller.yaml"),
+            "w",
+        ) as file:
+            yaml.dump(worker_poller_file, file, default_flow_style=False)
+
+    if traps:
+        # Load docker-compose-traps.yaml to dictionary and filter out secrets destined for deletion
+        with open(
+            os.path.join(path_to_compose_files, "docker-compose-traps.yaml")
+        ) as file:
+            traps_file = yaml.load(file, Loader=yaml.FullLoader)
+        traps_file["services"]["traps"]["secrets"] = list(
+            filter(
+                lambda el: el["source"] not in secrets,
+                traps_file["services"]["traps"]["secrets"],
+            )
+        )
+
+        # Save updated docker-compose-traps.yaml configuration
+        with open(
+            os.path.join(path_to_compose_files, "docker-compose-traps.yaml"), "w"
+        ) as file:
+            yaml.dump(traps_file, file, default_flow_style=False)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Manage secrets in docker compose")
+    parser.add_argument("--delete", default="false", help="If true, delete the secret")
+    parser.add_argument("--secret_name", help="Secret name")
+    parser.add_argument("--path_to_compose", help="Path to dockerfiles")
+    parser.add_argument(
+        "--worker_poller", default="true", help="Add secret to worker poller"
+    )
+    parser.add_argument("--traps", default="true", help="Add secret to traps")
+    parser.add_argument("--userName", default="", help="SNMPV3 username")
+    parser.add_argument("--privProtocol", default="", help="SNMPV3 privProtocol")
+    parser.add_argument("--privKey", default="", help="SNMPV3 privKey")
+    parser.add_argument("--authProtocol", default="", help="SNMPV3 authProtocol")
+    parser.add_argument("--authKey", default="", help="SNMPV3 authKey")
+    parser.add_argument("--contextEngineId", default="", help="SNMPV3 contextEngineId")
+
+    args = parser.parse_args()
+
+    # Assign inputs from command line to variables
+    delete_secret = human_bool(args.delete)  # flag indicating whether secret should be created or deleted
+    secret_name = args.secret_name  # name of the secret
+    path_to_compose_files = args.path_to_compose  # path to the directory with .env and docker-compose.yaml files
+    worker_poller = human_bool(args.worker_poller)  # flag indicating whether changes should be made in worker poller service
+    traps = human_bool(args.traps)  # flag indicating whether changes should be made in traps service
```

Comment:
I think for this part we can remove comments, as the variable names should imply what are they. I think for `secret_name` and `path_to_compose_files` is ok, but the names for flags could be better, to point that they are flags to concrete action, for example `do_change_for_worker_poller`.